app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SearchController.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.forms.{SearchForm, SearchFormData}
+import uk.gov.hmrc.tariffclassificationfrontend.models.Sort.Sort
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Search, Sort}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views.html
+
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class SearchController @Inject()(authenticatedAction: AuthenticatedAction,
+                                 casesService: CasesService,
+                                 val messagesApi: MessagesApi,
+                                 implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
+
+  def search(reference: Option[String] = None, search: Search = Search(), sort: Sort = Sort.COMMODITY_CODE): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    if (reference.isDefined) {
+      successful(Redirect(routes.CaseController.trader(reference.get)))
+    } else if (search.isEmpty) {
+      Future.successful(Results.Ok(html.advanced_search(fillForm(search), None)))
+    } else {
+      casesService.search(search, sort) map { results =>
```

Comment:
are you using implicitly an execution context here? if so, make sure you use the correct one, not the deprecated

Reply:
Nothing in the logs so I presume its using the new one

Reply:
Just checked with `cmd + shift + p`: you are using the default one (`mdcExecutionContext`) which is deprecated

please add this:
```
import scala.concurrent.ExecutionContext.Implicits.global
```

Reply:
Fixed